src/python/grpcio_tests/tests_py3_only/interop/xds_interop_client_test.py

```diff
@@ -0,0 +1,148 @@
+
+_CLIENT_PATH = os.path.abspath(os.path.realpath(xds_interop_client.__file__))
+_SERVER_PATH = os.path.abspath(os.path.realpath(xds_interop_server.__file__))
+
+_METHODS = (
+    (messages_pb2.ClientConfigureRequest.UNARY_CALL, "UNARY_CALL"),
+    (messages_pb2.ClientConfigureRequest.EMPTY_CALL, "EMPTY_CALL"),
+)
+
+_QPS = 100
+_NUM_CHANNELS = 20
+
+_TEST_ITERATIONS = 10
+_ITERATION_DURATION_SECONDS = 1
+_SUBPROCESS_TIMEOUT_SECONDS = 2
+
+
+@contextlib.contextmanager
+def _start_python_with_args(
+    file: str, args: List[str]
+) -> Tuple[subprocess.Popen, tempfile.TemporaryFile, tempfile.TemporaryFile]:
+    with tempfile.TemporaryFile(mode='r') as stdout:
+        with tempfile.TemporaryFile(mode='r') as stderr:
+            proc = subprocess.Popen((sys.executable, file) + tuple(args),
+                                    stdout=stdout,
+                                    stderr=stderr)
+            yield proc, stdout, stderr
+
+
+def _dump_stream(process_name: str, stream_name: str,
+                 stream: tempfile.TemporaryFile):
+    sys.stderr.write(f"{process_name} {stream_name}:\n")
+    stream.seek(0)
+    sys.stderr.write(stream.read())
+
+
+def _dump_streams(process_name: str, stdout: tempfile.TemporaryFile,
+                  stderr: tempfile.TemporaryFile):
+    _dump_stream(process_name, "stdout", stdout)
+    _dump_stream(process_name, "stderr", stderr)
+    sys.stderr.write(f"End {process_name} output.\n")
+
+
+def _test_client(server_port: int, stats_port: int, qps: int,
+                 num_channels: int):
+    settings = {
+        "target": f"localhost:{stats_port}",
+        "insecure": True,
+    }
+    for i in range(_TEST_ITERATIONS):
+        target_method, target_method_str = _METHODS[i % len(_METHODS)]
+        test_pb2_grpc.XdsUpdateClientConfigureService.Configure(
+            messages_pb2.ClientConfigureRequest(types=[target_method]),
+            **settings)
+        response = test_pb2_grpc.LoadBalancerStatsService.GetClientAccumulatedStats(
+            messages_pb2.LoadBalancerAccumulatedStatsRequest(), **settings)
+        before = {}
+        for _, method_str in _METHODS:
+            before[method_str] = response.stats_per_method[method_str].result[0]
+        time.sleep(_ITERATION_DURATION_SECONDS)
+        response = test_pb2_grpc.LoadBalancerStatsService.GetClientAccumulatedStats(
+            messages_pb2.LoadBalancerAccumulatedStatsRequest(), **settings)
+        after = {}
+        delta = {}
+        for _, method_str in _METHODS:
+            after[method_str] = response.stats_per_method[method_str].result[0]
+            delta[method_str] = after[method_str] - before[method_str]
+        logging.info("Delta: %s", delta)
+        for _, method_str in _METHODS:
+            if method_str == target_method_str:
+                assert delta[method_str] > 0, delta
+            else:
+                assert delta[method_str] == 0, delta
```

Comment:
optional: We can also check the status code distribution and other fields in the response. This is the first unit test (end2end unit test) around the Python xDS interop client, we can use it to validate the counting logic is correct. In case, in future, someone else (like me) want to change the statistic logic.

Reply:
Done.